module/gui/Options.py

```diff
@@ -0,0 +1,1500 @@
+            self.adjustSize()
+        return QDialog.exec_(self)
+
+    def defaultSettings(self):
+        self.settings.clear()
+        self.cbEnableFileLog.setChecked(False)
+        self.leFolder.setText("Logs")
+        self.cbRotate.setChecked(True)
+        self.sbSize.setValue(100)
+        self.sbCount.setValue(5)
+        self.cbException.setChecked(True)
+        self.dialogState2dict()
+
+    def dialogState2dict(self):
+        self.settings["file_log"]   = self.cbEnableFileLog.isChecked()
+        self.settings["log_folder"] = unicode(self.leFolder.text())
+        self.settings["log_rotate"] = self.cbRotate.isChecked()
+        self.settings["log_size"]   = self.sbSize.value()
+        self.settings["log_count"]  = self.sbCount.value()
+        self.settings["exception"]  = self.cbException.isChecked()
+
+    def dict2dialogState(self):
+        self.cbEnableFileLog.setChecked (self.settings["file_log"])
+        self.leFolder.setText           (self.settings["log_folder"])
+        self.cbRotate.setChecked        (self.settings["log_rotate"])
+        self.sbSize.setValue            (self.settings["log_size"])
+        self.sbCount.setValue           (self.settings["log_count"])
+        self.cbException.setChecked     (self.settings["exception"])
+
+    def appFontChanged(self):
+        self.buttons.updateWhatsThisButton()
+
+class ClickNLoadForwarderOptions(QDialog):
+    """
+        ClickNLoad port forwarder options dialog
+    """
+    def __init__(self, parent):
+        QDialog.__init__(self, parent)
+        self.log = logging.getLogger("guilog")
+
+        self.settings = {}
+        self.lastFont = None
+
+        self.setAttribute(Qt.WA_DeleteOnClose, False)
+        self.setWindowFlags(self.windowFlags() & ~Qt.WindowContextHelpButtonHint)
+        self.setWindowTitle(_("Options"))
+        self.setWindowIcon(QIcon(join(pypath, "icons", "logo.png")))
+
+        self.cbEnable = QCheckBox(_("Enable"))
+        self.lblFrom = QLabel(_("Local Port"))
+        self.sbFromPort = QSpinBox()
+        self.sbFromPort.setMinimum(1)
+        self.sbFromPort.setMaximum(65535)
+        lblTo = QLabel(_("Remote Port"))
+        self.sbToPort = QSpinBox()
+        self.sbToPort.setMinimum(1)
+        self.sbToPort.setMaximum(65535)
+        self.cbGetPort = QCheckBox(_("Get Remote Port from Server Settings"))
+        lblSta = QLabel(_("Status"))
+        self.lblStatus = LineView("Unknown")
+        self.lblStatus.setAlignment(Qt.AlignHCenter)
+
+        whatsThis = whatsThisFormat(self.lblFrom.text(), _("This should usually stay on port 9666. At least Firefox with FlashGot only works reliable with its default address setting:<br>http://127.0.0.1:9666/flashgot"))
+        self.lblFrom.setWhatsThis(whatsThis)
+        self.sbFromPort.setWhatsThis(whatsThis)
+        whatsThis = whatsThisFormat(self.cbGetPort.text(), _("Needs") + " '" + _("Settings") + "'" + " (SETTINGS) " + _("permission on the server."))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['_' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)